scrapy/utils/reqser.py

```diff
@@ -4,15 +4,17 @@
 import inspect
 
 from scrapy.http import Request
+from scrapy.spiders import Spider
 from scrapy.utils.python import to_unicode
 from scrapy.utils.misc import load_object
 
 
-def request_to_dict(request, spider=None):
-    """Convert Request object to a dict.
+def request_to_dict(request: Request, spider: Spider = None):
```

Comment:
```suggestion
def request_to_dict(request: Request, spider: Optional[Spider] = None) -> dict:
```

this would need Optional import. See also: https://github.com/python/mypy/issues/9091